[Cogs Face] Update Face SDK with latest auto-rest Swagger and update unit tests

lebronJ opened the pull request:
- Regenerate Face SDK with latest spec Azure/azure-rest-api-specs#2305
- Update unit tests (regenerate unit-test session records)
- Increment package version to `1.0.2`
- Divide `ComputerVision` and `Face` aligning with other Vision SDKs.
  
This has been internally reviewed (lebronJ/azure-sdk-for-net#1) inside Face SDK team.

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj:
```diff
@@ -3,7 +3,7 @@
     <PackageId>Microsoft.Azure.CognitiveServices.Vision</PackageId>
     <Description>This client library provides access to the Microsoft Cognitive Services Vision APIs.</Description>
-    <VersionPrefix>1.0.1-preview</VersionPrefix>
+    <VersionPrefix>1.0.2-preview</VersionPrefix>
```
@lebronJ please model your csproj according to the below project.
https://github.com/Azure/azure-sdk-for-net/blob/psSdkJson6/src/SDKs/CognitiveServices/dataPlane/Search/BingCustomSearch/BingCustomSearch/Microsoft.Azure.CognitiveServices.Search.CustomSearch.csproj
you are missing imports at the bottom of the project file and also you don't need few project properties that you have added.

lebronJ replied:
Refined this csproj following the styles of above `BingCustomSearch`

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision.Tests/Face/BaseTests.cs:
```diff
@@ -0,0 +1,35 @@
+        {
+            // Retrieve the configuration information.
+            FaceSubscriptionKey = "";
+            Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Playback");
```
@lebronJ you do not set this in test case, rather you set this in your connection string.
you use test framework to record tests
https://github.com/Azure/azure-powershell/blob/preview/documentation/Using-Azure-TestFramework.md#manually-set-environment-variables

lebronJ replied:
Removed this unnecessary setting
`Environment.SetEnvironmentVariable("AZURE_TEST_MODE", "Playback");`

shahabhijeet reviewed (changes requested):
Please sync with David li in case you need more information about recording tests. If not let me know if you have any questions. I am tagging David on this PR

shahabhijeet commented:
@DavidLiCIG can you review the test part of this PR

lebronJ commented:
@shahabhijeet Updated following your comments.
If this is fine to merge, could you please help share how to release this updated Nuget package? Thanks.

DavidLiCIG commented:
This package should be released as an update to Microsoft.Azure.CognitiveServices.Vision package which is already on version 1.0.1.preview.

shahabhijeet commented on src/SDKs/CognitiveServices/dataPlane/Vision/Vision/Microsoft.Azure.CognitiveServices.Vision.csproj:
```diff
@@ -3,20 +3,15 @@
-    <Folder Include="Generated\Face\Models\" />
-    <Folder Include="Properties\" />
-  </ItemGroup>
+  <Import Condition=" Exists('$([MSBuild]::GetPathOfFileAbove(AzSdk.RP.props))') " Project="$([MSBuild]::GetPathOfFileAbove('AzSdk.RP.props'))" />
```
@lebronJ add the following comment above/below line 2 and line 16

lebronJ replied:
Done

lebronJ commented:
@shahabhijeet Updated following your comments. Any updates? Thanks.

shahabhijeet commented:
Have you used generate.cmd to generate your SDK? Using generate.cmd is essential to create the meta data file (it's a .txt file that has information about from what this SDK was generated)
I see your missing that file.

lebronJ commented:
@shahabhijeet Updated by regenerating SDK with `generate.cmd`. Thanks for your comments and sorry for not noticing this before.

shahabhijeet reviewed (changes requested):
Please add your vision sdk under this directory
SDKs\CognitiveServices\dataPlane\Vision\VisionSdk\VisionSDKs\CognitiveServices\dataPlane\Vision\VisionSdk\Vision.Tests

This helps in identifying tests for your sdk.
This aligns with all the other SDKs that are checked in, e.g. ContentModerator

lebronJ commented:
@shahabhijeet I have updated this PR by separating `Computer Vision` / `Face` SDKs and aligning them with `ContentModerator` and other Cogs SDKs using the same folder structure.
SDKs are regenerated by `generate.cmd` after adjusting the `readme.md` in Azure/azure-rest-api-specs#2305.

shahabhijeet reviewed (changes requested):
Please remove CognitiveServices/dataPlane/Vision/Vision/Generated/Face
I believe those files were left when you moved the files.

At this point you are better off create a new PR, this PR is already riddled with lot of commits.
It's better you create a new PR with all these changes with minimum commits

lebronJ committed:
Regenerate Face/ComputerVision SDK with latest spec and update the unit tests, separate Face/CV aligning with other Vision APIs.

lebronJ commented:
Thanks @shahabhijeet , I removed the unused Vision SDK folder as you commented. 
Considering creating a new PR with minimal commits, I think it's not straightforward to separate this PR, because this PR only does two **interdependent** things:
1. Regenerate Face & CV SDK with latest spec in Azure/azure-rest-api-specs#2305, and update unit tests based on the SDK changes.
2. Separate Face & CV SDKs aligning with other Vision SDKs.

So I rebased this PR to latest commit and force pushed it as one single commit.

lebronJ commented:
Hi @shahabhijeet , could you please help share how to release this updated Nuget package? Thanks.
(Previously https://www.nuget.org/packages/Microsoft.Azure.CognitiveServices.Vision/)

One thing to note: 
As this PR totally separated `Computer Vision` and `Face` into two independent SDKs, I think we should release **TWO** Nuget packages `Microsoft.Azure.CognitiveServices.Vision.ComputerVision` and `Microsoft.Azure.CognitiveServices.Vision.Face`.

DavidLiCIG commented:
Hi @lebronJ you can use this job to publish the new Nuget packages.

https://azuresdkci.westus2.cloudapp.azure.com/view/NetCore-SDK/job/NetCore-SDK-Publish/

lebronJ commented:
Thanks @DavidLiCIG , published.
